src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -3357,14 +3357,33 @@ bool Unit::IsUnderWater() const
     return GetBaseMap()->IsUnderWater(GetPositionX(), GetPositionY(), GetPositionZ());
 }
 
-void Unit::UpdateUnderwaterState(Map* m, float x, float y, float z)
+void Unit::UpdatePositionData()
+{
+    PositionFullVMapData data;
```

Comment:
FullTerrainStatus (naming - matters here because you are returning GridMap data if on flat/nowmo terrain)